syntax = "proto3";

package bayesmix;

/*
 * Enum for the different types of Hierarchy.
 */
enum HierarchyId {
    UNKNOWN_HIERARCHY = 0;
    NNIG = 1; // Normal - Normal Inverse Gamma
    NNW = 2; // Normal - Normal Wishart
    LinRegUni = 3; // Linear Regression (univariate response)
    LapNIG = 4; // Laplace - Normal Inverse Gamma
    FA = 5; // Factor Analysers
    NNxIG = 6; // Normal - Normal x Inverse Gamma
    PythonHier = 7; // Generic python hierarchy
}
